app/src/main/java/com/github/sdp/mediato/data/UserDatabase.java

```diff
@@ -324,4 +249,37 @@ public static void setValueInFollowing(String myUsername, String targetUserUsern
                 .addOnCompleteListener(task -> System.out.println(myUsername + " is now set to " + value + " in " + targetUserUsername + " followers list."));
     }
 
+    /**
+     * Updates the user's location in the database
+     * @param username
+     * @param latitude
+     * @param longitude
+     */
+    public static void updateLocation(String username, double latitude, double longitude) {
+        Location location = new Location(latitude, longitude);
+        Preconditions.checkLocation(location);
+        database.getReference().child(USERS_PATH + username + LOCATION_PATH)
+                .setValue(location);
+    }

```

Comment:
Might it be a good idea if it returns a` CompletableFuture<Void>` ? For example, if we need to wait for the completion of the update before updating a view.

Reply:
I thought about it but at the moment I am still unsure about how / when the updating is going to happen. Once I figure that out (it's my next task), I might change the return type but it is not necessary yet at the moment (like for the others methods that update data).